File: 1651-shuffle-string/1651-shuffle-string.cpp
class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        // map<int,char> mp;
        // for(int i = 0; i < s.length(); i++)
        // {
        //     mp[indices[i]] = s[i];
        // }

        // string res;
        // for(auto itr = mp.begin(); itr != mp.end(); itr++)
        // {
        //     res += (itr->second);
            
        // }
        // return res;

         for(int i=0;i<indices.size();i++)
        {
            
              while(indices[i]!=i)
              {
                  swap(s[i],s[indices[i]]);
                  swap(indices[i],indices[indices[i]]);
                       
              }
        }
        
       
        return s;
        
    }
};

File: 1651-shuffle-string/1651-shuffle-string.java
class Solution {

    public static void swapArray(int indices[], int element, int index)
    {
        int temp = indices[index];
        indices[index] = indices[element];
        indices[element] = temp;

    }

    

    public String restoreString(String s, int[] indices) {

        StringBuilder sb = new StringBuilder("");
        char[] charray = new char[s.length()];

        for(int i = 0; i < indices.length; i++)
        {
            charray[indices[i]] = s.charAt(i);
        }

        // String str = "";

        // for(char i : charray)
        // {
        //     str += i;
        // }

        sb.append(charray);


        return sb.toString();

    }
}